src/main/java/net/geforcemods/securitycraft/entity/SentryEntity.java

```diff
@@ -165,8 +168,18 @@ else if(item == SCContent.DISGUISE_MODULE.get())
 				ItemStack module = getDisguiseModule();
 
 				if(!module.isEmpty()) //drop the old module as to not override it with the new one
+				{
 					Block.spawnAsEntity(world, getPosition(), module);
 
+					List<ItemStack> blocks = ((ModuleItem)module.getItem()).getAddons(module.getTag());
+					if(blocks.size() > 0)
+					{
+						BlockState state = Block.getBlockFromItem(blocks.get(0).getItem()).getDefaultState();
+						if(state == world.getBlockState(pos))
+							world.setBlockState(getPosition(), Blocks.AIR.getDefaultState());

```

Comment:
Use `world.destroyBlock`.

Reply:
You can hear a destroying sound when switching modules after that change. Should that be?

Reply:
Yes. I'm not a huge fan of blocks just poofing out of existence :P

Reply:
..and I noticed that for some reason the breaking sound is played twice.